test/test_configs/generate_test_configs.py

```diff
@@ -22,6 +22,16 @@ def get_config(model_config_path, storage_config_path, train_config_path, eval_c
 def set_dataset_config(base_config, dataset_dir):
     dataset_config_path = dataset_dir / Path("dataset.yaml")
     dataset_config = OmegaConf.load(dataset_config_path)
+    
+    # the below attributes need not be manually set as they will be automatically retrieved from dataset_config_path
+    dataset_config.num_edges = -1
```

Comment:
Is this code necessary if the defaults for the DatasetConfig are updated?

Reply:
this function is used to generate user configs in the test cases. But the input file being passed has stats on `num_edges`, `num_nodes` etc which we don't want to supply in the generated user config, hence we are explicitly setting them to -1. 

Doing this ensures that the user configs need not contain these values, and the tooling will retrieve them from `dataset_dir`. (since the tests go through, we can be sure that `num_edges` etc are being read from `dataset.yaml`)